lib/virtfs/xfs/file_methods.rb

```diff
@@ -0,0 +1,226 @@
+module VirtFS::XFS
+  class FS
+    module FileMethods
+
+      def file_atime(p)
+        de = get_file(p)
+        raise Errno::ENOENT, "No such file or directory #{p}" if de.nil?
+        superblock.inode(de.inode).access_time
+      end
+
+      def file_blockdev?(p)
+        de = get_file(p)
+        return false if de.nil?
+        superblock.inode(de.inode).mode_set?(VirtFS::Inode::FM_BLOCK_DEV)
+      end
+
+      def file_chardev?(p)
+        de = get_file(p)
+        return false if de.nil?
+        superblock.inode(de.inode).mode_set?(VirtFS::Inode::FM_CHAR)
+      end
+
+      def file_chmod(_permission, _p)
+        raise "VirtFS::XFS Write functionality is not yet supported on XFS"
+      end
+
+      def file_chown(_owner, _group, _p)
+        raise "VirtFS::XFS Write functionality is not yet supported on XFS"
+      end
+
+      def file_ctime(p)
+        de = get_file(p)
+        raise Errno::ENOENT, "No such file or directory #{p}" if de.nil?
+        superblock.inode(de.inode).create_time
+      end
+
+      def file_delete(_p)
+        raise "VirtFS::XFS Write functionality is not yet supported on XFS"
+      end
+
+      def file_directory?(p)
+        de = get_file(p)
+        return false if de.nil?
+        de.directory?
+      end
+
+      def file_executable?(p)
+        de = get_file(p)
+        return false if de.nil?
+        superblock.inode(de.inode).owner_permissions && VirtFS::Inode::PF_O_EXECUTE ||
+        superblock.inode(de.inode).group_permissions && VirtFS::Inode::PF_G_EXECUTE ||
+        superblock.inode(de.inode).user_permissions  && VirtFS::Inode::PF_U_EXECUTE
+      end
+
+      def file_executable_real?(p)
+      end
+
+      def file_exist?(p)
+        !get_file(p).nil?
+      end
+
+      def file_file?(p)
+        de = get_file(p)
+        !de.nil? && de.file?
+      end
+
+      def file_ftype(p)
+        de = get_file(p)
+        raise Errno::ENOENT, "No such file or directory #{p}" if de.nil?
+      end
+
+      def file_lchmod(_permission, _p)
+        raise "VirtFS::XFS Write functionality is not yet supported on XFS"
+      end
+
+      def file_lchown(_owner, _group, _p)
+        raise "VirtFS::XFS Write functionality is not yet supported on XFS"
+      end
+
+      def file_link(_p1, _p2)
+        raise "VirtFS::XFS Write functionality is not yet supported on XFS"
+      end
+
+      def file_lstat(p)
+        de = get_file(p)
+        raise Errno::ENOENT, "No such file or directory #{p}" if de.nil?
+        VirtFS::Stat.new(VirtFS::XFS::File.new(de, superblock).to_h)
+      end
+
+      def file_mtime(p)
+        de = get_file(p)
+        raise Errno::ENOENT, "No such file or directory #{p}" if de.nil?
+        superblock.inode(de.inode).modification_time
+      end
+
+      def file_new(p, parsed_args, _open_path, _cwd)
+        de = get_file(p)
+        raise Errno::ENOENT, "No such file or directory #{p}" if de.nil?
+        de
```

Comment:
The `file_new` method needs to return an object that implements the minimum methods required for the File instance thin interface. Does this directory entry object do that? It doesn't seem so, given the `file_size` method calls `superblock.inode(de.inode).length` instead of just calling `de.size`.

The object returned by this method is wrapped by the `ThinFileDelegator` of `VirtFS`. In the `ThinFileDelegator` object, it is accessed as `@fs_io_obj` and needs to implement all the methods the `ThinFileDelegator` expects of it.